### PR TITLE
LSP: Groovy diagnostic handling.

### PR DESCRIPTION
**Pull request by dbalek**
- ErrorProvider implementation for GSF languages added
- Groovy hints foixed to work over LSP

### CODE REVIEW

File: ide/csl.api/src/org/netbeans/modules/csl/core/ApiAccessor.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import java.util.List;
+import org.netbeans.api.annotations.common.NonNull;
+import org.netbeans.modules.csl.api.EditList;
+import org.openide.util.Parameters;
+
+/**
+ *
+ * @author Dusan Balek
+ */
+public abstract class ApiAccessor {
```
Comment (JaroslavTulach) [inline ide/csl.api/src/org/netbeans/modules/csl/core/ApiAccessor.java:30]: API without an accessor pattern is just like a cake without ketchup!

File: java/java.lsp.server/vscode/package.json [hunk 2]
```diff
@@ -358,5 +358,8 @@
 		"vscode-languageclient": "6.1.3",
 		"vscode-test-adapter-api": "^1.9.0",
 		"vscode-test-adapter-util": "^0.7.1"
-	}
+	},
+	"extensionDependencies": [
+		"hbenl.vscode-test-explorer"
```
Comment (JaroslavTulach) [inline java/java.lsp.server/vscode/package.json:363]: Great.

**Review by JaroslavTulach — APPROVED** [review]

**Review by jlahoda — APPROVED** [review]
Looks good to me.

File: java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/protocol/TextDocumentServiceImpl.java [hunk 3]
```diff
@@ -764,139 +758,120 @@ private DocumentSymbol element2DocumentSymbol(CompilationInfo info, Element el)
...
+                            for (Union2<org.netbeans.api.lsp.TextDocumentEdit, org.netbeans.api.lsp.ResourceOperation> parts : edit.getDocumentChanges()) {
+                                if (parts.hasFirst()) {
+                                    String docUri = parts.first().getDocument();
+                                    try {
+                                        FileObject file = Utils.fromUri(docUri);
+                                        if (file == null) {
+                                            file = Utils.fromUri(params.getTextDocument().getUri());
+                                        }
+                                        FileObject fo = file;
+                                        List<TextEdit> edits = parts.first().getEdits().stream().map(te -> new TextEdit(new Range(Utils.createPosition(fo, te.getStartOffset()), Utils.createPosition(fo, te.getEndOffset())), te.getNewText())).collect(Collectors.toList());
```
Comment (sdedic) [inline java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/protocol/TextDocumentServiceImpl.java:800]: Maybe guard against `fo == null` (unlikely, possible).

**Review by sdedic — APPROVED** [review]

**Review by svenreimers — APPROVED** [review]

File: java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/protocol/TextDocumentServiceImpl.java (hunk 3 above)
Comment (dbalek) [reply java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/protocol/TextDocumentServiceImpl.java]: Done.

**Pull request merged**